core/src/test/scala/neotypes/EffectSuite.scala

```diff
@@ -0,0 +1,47 @@
+package neotypes
+
+import neotypes.internal.syntax.StageSyntaxSpec
+import org.scalatest.{AsyncTestSuite, Suites}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+
+/** Testkit used to write specs abstracted from any concrete effect. */
+abstract class EffectTestkit[F[_]](implicit ct: ClassTag[F[_]]) {
+  final val effectName: String = ct.runtimeClass.getCanonicalName
+
+  trait Behaviour {
+    def fToFuture[T](f: F[T]): Future[T]
+    def asyncInstance: Async[F]
+  }
+
+  def createBehaviour(implicit ec: ExecutionContext): Behaviour
+}
+
+/** Base class for writing effect specs. */
+abstract class BaseEffectSpec[F[_]](testkit: EffectTestkit[F]) extends AsyncTestSuite { self =>
+  protected final val effectName: String =
+    testkit.effectName
+
+  private final val behaviour: testkit.Behaviour =
+    testkit.createBehaviour(self.executionContext)
+
+  protected final def fToFuture[T](f: F[T]): Future[T] =
+    behaviour.fToFuture(f)
+
+  protected implicit final val F: Async[F] =
+    behaviour.asyncInstance
+}
+
+/** Group all the effect specs into one big suite, which can be called for each effect. */
+abstract class EffectSuite[F[_]](testkit: EffectTestkit[F]) extends Suites(
```

Comment:
I like the unifying the tests for the effects in one place, and passing in the EffectTestkit[F] for the different effect types.  One question about this approach, how would I run just one of these tests?  I can forsee the use case where a developer is making changes to a specific part of the codebase, and just wants to run the test that hits that code and not the entire suite.

Reply:
Yeah you can simply create a `final class JustRunThis extends TheSpecYouWant(FutureTestkit)` or you can create your own suite, or you may comment the classes you do not want to run for the main suite.

In the end, you want to ensure your changes don't break anything.
So I am ok with the fact that just wanting to run a couple of specs requiring some extra work of the developer. Because, those changes will never be committed.

Reply:
Sounds good, makes sense.